### PR TITLE
Support parallelization for Downstream CI

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: openshift/e2e-tests-openshift.sh [hunk 1]
```diff
@@ -97,6 +103,87 @@ function in_array() {
     return 1
 }
 
+function test_privileged {
+    local cnt=0
+    local task_to_tests=""
+
+    # Run the non privileged tests
```
Comment (vinamra28) [inline openshift/e2e-tests-openshift.sh]: nit:
```suggestion
    # Run the privileged tests
```

File: openshift/e2e-tests-openshift.sh [hunk 2]
```diff
@@ -97,6 +103,87 @@ function in_array() {
...
+            task_to_tests=""
+            continue
+        fi
+
+        cnt=$((cnt+1))
+    done
+
+    # Remaining task
+    if [[ -n ${task_to_tests} ]];then
+        echo "Running non privileged test: ${task_to_tests}"
```
Comment (vinamra28) [inline openshift/e2e-tests-openshift.sh]:
```suggestion
        echo "Running privileged test: ${task_to_tests}"
```

**Review by vinamra28 — COMMENTED** [review]
a few nitpicks otherwise looks good to me :smiley:

**Review by vinamra28 — APPROVED** [review]
/cc @piyush-garg

File: openshift/e2e-tests-openshift.sh [hunk 3]
```diff
@@ -97,6 +103,91 @@ function in_array() {
...
+        function pre-apply-taskrun-hook() {
+            cp ${TMPF} ${TMPF2}
+            python3 openshift/e2e-add-service-account.py ${SERVICE_ACCOUNT} < ${TMPF2} > ${TMPF}
+            oc adm policy add-scc-to-user privileged system:serviceaccount:${tns}:${SERVICE_ACCOUNT} || true
+        }
+        unset -f pre-apply-task-hook || true
+
+        task_to_tests="${task_to_tests} task/${runtest}/*/tests"
+
+        if [[ ${cnt} == "${MAX_NUMBERS_OF_PARALLEL_TASKS}" ]];then
```
Comment (piyush-garg) [inline openshift/e2e-tests-openshift.sh:125]: Should this be `<=`

Comment (piyush-garg) [reply openshift/e2e-tests-openshift.sh:125]: or `!=`

File: openshift/e2e-tests-openshift.sh [hunk 4]
```diff
@@ -4,13 +4,37 @@
 #
 set -e
 
+# Maximin number of parallel tasks run at the same time
+# # start from 0 so 4 => 5
+MAX_NUMBERS_OF_PARALLEL_TASKS=4
```
Comment (chmouel) [inline openshift/e2e-tests-openshift.sh:9]: https://en.wikipedia.org/wiki/Zero-based_numbering

**Pull request merged**